contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+        uint256 feeAccumulated = (simpleFeeFactor * totalUtilizedSD) / DECIMAL;
+        uint256 totalUtilizedSDNew = feeAccumulated + totalUtilizedSD;
+        uint256 totalProtocolFeeNew = (protocolFeeFactor * feeAccumulated) / DECIMAL + accumulatedProtocolFee;
+        if (cTokenTotalSupply == 0) {
+            return DECIMAL;
+        } else {
+            uint256 poolBalancePlusUtilizedSDMinusReserves = getPoolAvailableSDBalance() +
+                totalUtilizedSDNew -
+                totalProtocolFeeNew;
+            uint256 exchangeRate = (poolBalancePlusUtilizedSDMinusReserves * DECIMAL) / cTokenTotalSupply;
+            return exchangeRate;
+        }
+    }
+
+    function getPoolAvailableSDBalance() public view override returns (uint256) {
+        return IERC20(staderConfig.getStaderToken()).balanceOf(address(this)) - sdReservedForClaim;
+    }
+
+    /// @notice Calculates the utilization of the utility pool
+    function poolUtilization() public view override returns (uint256) {
+        // Utilization is 0 when there are no utilized SD
+        if (totalUtilizedSD == 0) {
+            return 0;
+        }
+
+        return (totalUtilizedSD * DECIMAL) / (getPoolAvailableSDBalance() + totalUtilizedSD - accumulatedProtocolFee);
+    }
+
+    /**
+     * @notice Calculates and returns the user data for a given account
+     * @param account The address whose utilisation should be calculated
+     * @return UserData struct containing the user data
+     */
+    function getUserData(address account) public view override returns (UserData memory) {
+        address staderOracle = staderConfig.getStaderOracle();
+        uint256 sdPriceInEth = IStaderOracle(staderOracle).getSDPriceInETH();
+        uint256 totalInterestSD = getUtilizerLatestBalance(account) -
+            ISDCollateral(staderConfig.getSDCollateral()).operatorUtilizedSDBalance(account);
+
+        uint256 totalCollateralInEth = getOperatorTotalEth(account);
+        uint256 totalCollateralInSD = totalCollateralInEth / sdPriceInEth;
+
+        uint256 healthFactor = (totalInterestSD == 0)
+            ? type(uint256).max
+            : (totalCollateralInSD * riskConfig.liquidationThreshold) / totalInterestSD;
+
+        return
+            UserData(
+                totalInterestSD,
+                totalCollateralInSD,
+                healthFactor,
+                liquidationIndexByOperator[account] == 0
+                    ? 0
+                    : liquidations[liquidationIndexByOperator[account] - 1].totalAmountInEth
+            );
+    }
+
+    /**
+     * @notice
+     * @param operator Calculates and returns the conservative estimate of the total Ether (ETH) bonded by a given operator.
+     * @return totalEth The total ETH bonded by the operator
+     */
+    function getOperatorTotalEth(address operator) public view returns (uint256) {
+        (, , uint256 totalValidators) = ISDCollateral(staderConfig.getSDCollateral()).getOperatorInfo(operator);
```

Comment:
Could we call this as nonTerminalKeys please?

These are keys/validators that haven't exited. Here `totalValidators` can be misinterpreted as the total historical count of validators (some of which could have exited too!)